packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -388,9 +388,11 @@ bool _needsCurlyBracketStringInterpolation(String messageString, String placehol
     // Examples:
     // "'The number of {hours} elapsed is: 44'" // no curly brackets.
     // "'哈{hours}哈'" // no curly brackets.
+    // "'#placeholder#placeholder#'" // no curly brackets
     // "'m#hours#m'" // curly brackets.
     // "'I have to work _#hours#_' sometimes." // curly brackets.
-    final RegExp commonCaseRE = RegExp('[^a-zA-Z_][#{]$placeholder[#}][^a-zA-Z_]');
+    final RegExp commonCaseRE =
+        RegExp('([^a-zA-Z_]?(\$(.*){)?[#{]$placeholder[#}][^a-zA-Z_])');
```

Comment:
I am worried about the new `(.*)`. Can you remind me what `.` accepts? What is the intention there?

Reply:
I don't remember exactly to why. But, it's matches all characters of placeholders.

Reply:
I think that this is too general. Could you make a more specific regex pattern than `.*`?